packages/monitors/index.js

```diff
@@ -80,201 +85,205 @@ async function run({
 
     const getTime = () => Math.round(new Date().getTime() / 1000);
 
-    const [detectedContract, networkId, latestBlock] = await Promise.all([
-      findContractVersion(financialContractAddress, web3),
-      web3.eth.net.getId(),
-      web3.eth.getBlock("latest")
-    ]);
-
-    // Append the contract version and type to the monitorConfig, if the monitorConfig does not already contain one.
-    if (!monitorConfig) monitorConfig = {};
-    if (!monitorConfig.contractVersion) monitorConfig.contractVersion = detectedContract.contractVersion;
-    if (!monitorConfig.contractType) monitorConfig.contractType = detectedContract.contractType;
-
-    // Check that the version and type is supported. Note if either is null this check will also catch it.
-    if (
-      SUPPORTED_CONTRACT_VERSIONS.filter(
-        vo => vo.contractType == monitorConfig.contractType && vo.contractVersion == monitorConfig.contractVersion
-      ).length == 0
-    )
-      throw new Error(
-        `Contract version specified or inferred is not supported by this bot. Monitor config:${JSON.stringify(
-          monitorConfig
-        )} & detectedContractVersion:${JSON.stringify(detectedContract)} is not part of ${JSON.stringify(
-          SUPPORTED_CONTRACT_VERSIONS
-        )}`
+    // List of promises to run in parallel during each iteration,
+    // each of which represents a monitor executing.
+    let monitorRunners = [];
+
+    /** *************************************
+     *
+     * Financial Contract Runner
+     *
+     ***************************************/
+    if (financialContractAddress) {
```

Comment:
@chrismaree as discussed IRL here we can run both OO and EMP monitors, depending on which addresses are specified